In kernel 4.17, pci_get_bus_and_slot() has been removed in favor of
pci_get_domain_bus_and_slot(). It appears that VirtualBox only uses
domain number 0, thus pci_get_domain_bus_and_slot(0, bus, devfn)
is a suitable replacement for pci_get_bus_and_slot(bus, devfn).

The resulting code compiles; however, I do not use PCI passthru, which
I think means that I have not actually tested the code.

This patch released under a combined MIT/GPLv2 license.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>

--- VirtualBox-kmod-5.2.10/vboxpci/linux/VBoxPci-linux.c.orig	2018-04-26 00:28:06.691593982 +0100
+++ VirtualBox-kmod-5.2.10/vboxpci/linux/VBoxPci-linux.c	2018-04-26 00:28:32.171875664 +0100
@@ -89,7 +89,14 @@ MODULE_VERSION(VBOX_VERSION_STRING " r"
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 20)
 # define PCI_DEV_GET(v,d,p)            pci_get_device(v,d,p)
 # define PCI_DEV_PUT(x)                pci_dev_put(x)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 17, 0)
+/* assume the domain number to be zero - exactly the same assumption of
+ * pci_get_bus_and_slot()
+ */
+# define PCI_DEV_GET_SLOT(bus, devfn)  pci_get_domain_bus_and_slot(0, bus, devfn)
+#else
 # define PCI_DEV_GET_SLOT(bus, devfn)  pci_get_bus_and_slot(bus, devfn)
+#endif
 #else
 # define PCI_DEV_GET(v,d,p)            pci_find_device(v,d,p)
 # define PCI_DEV_PUT(x)                do { } while (0)
